### PR TITLE
[JENKINS-54426] Fix failures in Java 11

### PR DESCRIPTION
**Pull request by varyvol**
[JENKINS-54426](https://issues.jenkins-ci.org/browse/JENKINS-54426)

Based in #41 

- Use platform classloader when JDK 9+ to be able to access platform classes.
- Include a demo that uses a base image with JDK11.

### CODE REVIEW

File: demo/cwp-jdk11/Makefile [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	docker build -t ${JENKINSFILE_RUNNER_TAG} -f $(shell pwd)/out/tmp/output/Dockerfile $(shell pwd)/out/tmp/output/	
+
+build: .copySource .build .buildImage
+
+# Slow, but it does not need anything except Docker
+buildInDocker: .copySource .buildInDocker .buildImage
+
+run:
+	docker run --rm -v $(shell pwd)/Jenkinsfile:/workspace/Jenkinsfile \
+	-e JAVA_OPTS="-p /usr/share/jenkins/ref/java_cp/jaxb-api.jar:/usr/share/jenkins/ref/java_cp/javax.activation.jar --add-modules java.xml.bind,java.activation,java.sql -cp /usr/share/jenkins/ref/java_cp/jaxb-impl.jar:/usr/share/jenkins/ref/java_cp/jaxb-core.jar" \
```
Comment (varyvol) [inline demo/cwp-jdk11/Makefile:49]: $JAVA_OPTS to be provided by the jenkinsfile-runner binary to the Java command.

File: demo/cwp-jdk11/packager-config.yml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    source:
+      version: "2.30"
+  - groupId: "org.jenkins-ci.plugins.workflow"
+    artifactId: "workflow-step-api"
+    source:
+      version: "2.14"
+  - groupId: "org.jenkins-ci.plugins.workflow"
+    artifactId: "workflow-support"
+    source:
+      version: "3.0" # TO-DO: Maybe this should be into payload-dependencies ?
```
Comment (varyvol) [inline demo/cwp-jdk11/packager-config.yml:45]: It's not in the ["base demo"](https://github.com/jenkinsci/jenkinsfile-runner/blob/master/demo/cwp/packager-config.yml) so probably should be ported to payload-dependencies module in a future JDK11 branch.

**Review by jvz — APPROVED** [review]
I find it amusing that you use Makefiles instead of Ant (the slightly better make but with more angles in the syntax).

File: demo/cwp-jdk11/Makefile (hunk 1 above)
Comment (oleg-nenashev) [reply demo/cwp-jdk11/Makefile:49]: Ideally we need to include it into the image, but we need to update CWP to do that (trivial change)

**Review by oleg-nenashev — APPROVED** [review]
So in #41 I was patching a wrong classloader, and then did not experiment with the proper one. Thanks @varyvol ! 

I think we need to somehow automate bundling JAVA_OPTS into the image, but it can be done later as a part of https://issues.jenkins-ci.org/browse/JENKINS-54906

File: demo/cwp-jdk11/packager-config.yml (hunk 2 above)
Comment (oleg-nenashev) [reply demo/cwp-jdk11/packager-config.yml:45]: I think we can update the payload in both branches. Better to have a new major release in place OOTB

File: demo/cwp-jdk11/Makefile (hunk 1 above)
Comment (varyvol) [reply demo/cwp-jdk11/Makefile:49]: https://issues.jenkins-ci.org/browse/JENKINS-54906 for that.

**Review by fcojfernandez — APPROVED** [review]
Looks great

**Comment by oleg-nenashev** [comment]
Thanks @varyvol !

**Pull request merged**